### ISSUE TITLE
Scene Restart Fails

### ISSUE THREAD

**Issue by cgfarmer4** [Unity-Technologies/experimental-ARInterface#43]
Looks like [this commit ](https://github.com/Unity-Technologies/experimental-ARInterface/commit/08cd8cfad726d1adc3a8d4fd87d313ec60ad1c5d)breaks the ARCore StopService() function. It appears that this was copied from the ARKitInterace functionality and I believe setting the entire object as null should also null its properties. That fixes the crash in my case. 

If there are no other known issues with updating this, I will file a PR.

```
        public override void StopService()
        {
            var anchors = m_Anchors.Keys;
            foreach (var anchor in anchors)
            {
                DestroyAnchor(anchor);
            }

            m_ARCoreSession.enabled = false;
            TextureReader_destroy();
            BackgroundRendering = false;
            // m_BackgroundRenderer.backgroundMaterial = null;
            // m_BackgroundRenderer.camera = null;
            m_BackgroundRenderer = null;
            IsRunning = false;
        }

```

Not sure if related but I can't seem to ever get the scene to restart after doing this.

**Comment by cgfarmer4** [comment]
@iBicha this seems related to your PR, https://github.com/Unity-Technologies/experimental-ARInterface/pull/28. Can you comment as to what needs to happen here?

**Comment by iBicha** [comment]
@cgfarmer4 technically, even if pausing/restarting features could work, they are technically not fully supported before ARCore 1.0, and definitely where not mapped out in this interface.

https://github.com/google-ar/arcore-unity-sdk/issues/79

With the new API released, there is an update to be worked, that is all.

Specifically, in the new `ARCoreSession.cs` script, we can see 4 obvious api calls:
```
/// <summary>
/// Unity Awake.
/// </summary>
public void Awake()
{
    LifecycleManager.Instance.CreateSession(this);
}

/// <summary>
/// Unity OnDestroy.
/// </summary>
public void OnDestroy()
{
    LifecycleManager.Instance.DestroySession();
}

/// <summary>
/// Unity OnEnable.
/// </summary>
public void OnEnable()
{
    LifecycleManager.Instance.EnableSession();
}

/// <summary>
/// Unity OnDisable.
/// </summary>
public void OnDisable()
{
    LifecycleManager.Instance.DisableSession();
}
```

And they simply need to be called from within the interface.

**Comment by cgfarmer4** [comment]
@iBicha gotcha ... I'll leave those parts commented out until you find a solution that works best with the new background API. Thanks for the reference link. Will look into that!

**Comment by cgfarmer4** [comment]
@iBicha I checked to see if those Lifecycle functions are being called in the current ARCoreInterface and since the object is enable/disabled, they are. I see you changed the UpdateCamera function and that fixes the crash associated with the comments here.

Have you been able to get the scene to restart? Still having trouble here. Have tried many different things and it seems all the OnEnable / OnDisable functions are firing as proper.

**Comment by iBicha** [comment]
I got the chance to check the latest changes on the interface, and it should be working properly.
How are you restarting the session? Stopping the session and starting it again should be as simple as enabling and disabling the `ARController` component.
The component is sitting on a game object called `ARRoot` in the examples, which as a `DontDestroyOnLoad` script. If you are doing a scene load to restart your scene, the controller is not affected. make sure to disable it and enable it again.

(as a quick test, I opened the `ARBallz` example, added a button to it, and made it execute this on click:
```
public void Toggle () {
        var controller = FindObjectOfType<ARController>();
        controller.enabled = !controller.enabled;
}
```
And I think it works as I expect it to be.
